Added property 'goToInverval' for Carousel

kindeep opened the pull request:
Longer transitions can be nauseating, using a shorter duration allows
this to look better.

Plus if someone really wants a longer transition.

gutiguy commented on src/components/Carousel.tsx:
```diff
@@ -3,7 +3,7 @@ import styled from "@emotion/styled";
 import leftNavigation from "../static/LeftNavigation.png";
 import rightNavigation from "../static/RightNavigation.png";
-import PropTypes from "prop-types";
+import PropTypes, { number } from "prop-types";
```
I'd prefer not to import `{ number }` from prop-types and just use `PropTypes.number` instead (in line 73), both to avoid confusion with the typescript number type and to stay consistent with the other usages of PropTypes.

kindeep replied:
Got rid of this

gutiguy commented on src/components/Carousel.tsx:
```diff
@@ -51,7 +53,8 @@ class Carousel extends Component<IProps, IState> {
     prevPropsGoToSlide: 0,
-    newSlide: false
+    newSlide: false,
+    goToInterval: GOTO_INTERVAL,
```
Is there any reason to keep `goToInterval` in the state? Seems to me like it can simply be a prop. Also I'd consider renaming it to `goToSlideDelay` (or something like that) since it's actually responsible for creating a delay between changing the slides. `goToInterval` could be confused to mean the duration of the entire movement taking that number of milliseconds.

kindeep replied:
No reason, got rid of it.

gutiguy commented on src/components/Carousel.tsx:
```diff
@@ -66,19 +69,21 @@ class Carousel extends Component<IProps, IState> {
 
     if (goToSlide !== state.prevPropsGoToSlide) {
-      return { prevPropsGoToSlide: goToSlide, goToSlide, newSlide: true };
+      return { prevPropsGoToSlide: goToSlide, goToSlide, newSlide: true, goToInterval };
```
No need to return `goToInterval` here (this comment is moot when you change it to just be a prop).

gutiguy commented on src/components/Carousel.tsx:
```diff
@@ -66,19 +69,21 @@ class Carousel extends Component<IProps, IState> {
     offsetRadius: 2,
-    animationConfig: { tension: 120, friction: 14 }
+    animationConfig: { tension: 120, friction: 14 },
+    goToInterval: GOTO_INTERVAL
```
I'd change `GOTO_INTERVAL` to `DEFAULT_GOTO_SLIDE_DELAY` for clarity.

kindeep replied:
Makes sense, also changed goToInterval to goToSlideDelay

kindeep commented:
Thank you for the great feedback! 

I've updated the pull request, changed goToInterval to goToSlideDelay, and only using it as a prop.

Also `GO_TO_INTERVAL` is now accordingly `DEFAULT_GO_TO_SLIDE_DELAY`

Let me know if any other changes are needed.

I was very new to react when I made the pull request, still fairly new, thanks for putting up with me!

gutiguy commented on README.md:
```diff
@@ -81,5 +81,6 @@ And you're all set. You can also use props for better control of how the carouse
 | --------------- | -------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | slides          | ---                              | An array containing elements of the form `{ key, content, onClick? }` where key holds any unique value and content holds a reference to a JSX `<img />` to be displayed. onClick is an optional property that may be used to pass a callback method which will be invoked when the slide is clicked. |
 | goToSlide       | `null`                           | Setting this prop will cause the carousel to animate towards the provided index in the slides array. If `showNavigation` is set to false this prop is how you're expected to control the carousel.                                                                                                   |
+| goToSlideDelay  | `200`                            | Setting this prop will change how long the carousel takes to animate towards the index provided in `goToSlide`.               |
```
```suggestion
| goToSlideDelay  | `200`                            | The amount of milliseconds to delay each slide before moving to the next one while animating towards the index provided in `goToSlide`.               |
```
